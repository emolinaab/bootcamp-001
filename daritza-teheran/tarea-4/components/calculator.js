export class Calculator{
    constructor(currentNumber) {
        this.current = currentNumber;
      }

      display(s){
        console.log(currentNumber)
      }
}