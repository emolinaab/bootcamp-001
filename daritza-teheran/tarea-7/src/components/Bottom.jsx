import image from "../assets/codigo-de-barras.png";

export default function Bottom({ data }) {
  return (
    <footer>
      <p>{data}</p>
      <img src={image} />
    </footer>
  );
}
