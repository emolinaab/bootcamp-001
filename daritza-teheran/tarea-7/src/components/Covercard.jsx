import PropTypes from "prop-types";
import "../styles/coverCard.css";

const listItems = (lines) => {
  return lines.map((line) => <li key={line.toString()}>{line}</li>);
};

export const CovercardLine = ({ lines }) => {
  return <ul>{listItems(lines)}</ul>;
};

export const CovercardImage = ({ url, lines }) => {
  return (
    <ul>
      <img src={url} />
      {listItems(lines)}
    </ul>
  );
};

export const CovercardBottom = ({ url, data }) => {
  return (
    <footer>
      <p>{data}</p>
      <img src={url} />
    </footer>
  );
};

CovercardLine.propTypes = {
  lines: PropTypes.array,
};

CovercardImage.propTypes = {
  lines: PropTypes.array,
  url: PropTypes.string,
};

CovercardBottom.propTypes = {
  data: PropTypes.string,
  url: PropTypes.string,
};
