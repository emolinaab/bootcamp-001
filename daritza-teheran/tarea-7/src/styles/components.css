h1 {
  color: white;
  font-size: 10vh;
  text-align: center;
  margin: 0;
  line-height: 0.65;
  padding: 15px;
}

h2 {
  color: #cc8334;
  font-size: 2.5vh;
  text-align: start;
  margin: 0;
  text-indent: 6.5vh;
  font-weight: 500;
  font-style: italic;
}

.cover-lines {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 65vh;
  width: 40vh;
}

.cover-lines > ul {
  list-style: none;
  color: #fafaf9;
  font-weight: 400;
  font-size: 2.7vh;
}

.cover-lines ul:first-child {
  margin-block-start: 10vh;
  margin-bottom: 10vh;
}

li:nth-child(2) {
  color: #cc8334;
}

ul:last-child {
  display: flex;
  flex-direction: column;
  height: 12vh;
  width: 68vh;
  align-items: flex-end;
}

ul img {
  display: block;
  height: 20h;
  width: 20vh;
  border-radius: 50%;
}

footer {
  display: flex;
  position: absolute;
  bottom: 0;
  height: 5vh;
  width: 17vh;
  padding: 5vh;
  background-color: #e9e916;
  font-weight: 400;
  font-size: 2vh;
  align-items: center;
  flex-direction: column;
  justify-content: space-evenly;
  color: black;
}

footer img {
  display: block;
  height: 6vh;
  width: 12vh;
}
