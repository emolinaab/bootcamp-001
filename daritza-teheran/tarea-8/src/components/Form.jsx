import { Button, Grid, Snackbar, TextField } from "@mui/material";
import { useState } from "react";
import useForm from "../utils/useForm";
import validateSubmit from "../utils/validateSubmit";

export const Form = ({ vSubmit = validateSubmit}) => {
  const { email, password, formState, onInputChange, resetSignIn } = useForm();
  const [{ open, msg }, setOpen] = useState({ open: false, msg: "" });

  const handleSubmit = (e) => {
    e.preventDefault();
    let errMsg = vSubmit(formState);
    if (errMsg === "Succesfully log in!") resetSignIn();
    setOpen({ open: true, msg: errMsg });
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Snackbar
        open={open}
        autoHideDuration={7000}
        onClose={handleClose}
        message={msg}
        ContentProps={{
          sx: {
            background: "#262254",
          },
        }}
      />
      <form data-testid="form">
        <Grid container>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              inputProps={{ 'aria-label': 'email' }}
              label="Email"
              type="email"
              placeholder="email@globant.com"
              value={email}
              onChange={onInputChange}
              required
              fullWidth
            />
          </Grid>

          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              inputProps={{ 'aria-label': 'password' }}
              label="Password"
              type="password"
              placeholder="Password"
              value={password}
              onChange={onInputChange}
              required
              fullWidth
            />
          </Grid>

          <Grid
            container
            justifyContent="center"
            spacing={2}
            sx={{ mb: 2, mt: 1 }}
          >
            <Grid item xs={12} sm={12}>
              <Button onClick={handleSubmit} variant="contained" fullWidth>
                Login
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </form>
    </>
  );
};
