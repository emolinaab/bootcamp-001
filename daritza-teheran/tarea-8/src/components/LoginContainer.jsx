import React from "react";
import { Grid, Typography } from "@mui/material";

export const LoginContainer = ({ children }) => {
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justifyContent="center"
      sx={{ minHeight: "100vh", backgroundColor: "secondary.main", padding: 4 }}
    >
      <Grid
        item
        className="'box-shadow"
        sx={{
          width: "60vh",
          backgroundColor: "white",
          padding: 3,
          borderRadius: 2,
        }}
      >
        <Typography variant="h5" sx={{ mb: 1 }}>
          Sign in
        </Typography>
        {children}
      </Grid>
    </Grid>
  );
};
