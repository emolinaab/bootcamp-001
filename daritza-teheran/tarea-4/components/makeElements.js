import { Calculator } from "./calculator.js";

const root = document.getElementById("root");
const current = document.querySelector('results')
const calculator = new Calculator(current)

export function makeSection(props) {
    const section = document.createElement("section");
    section.setAttribute("id", props);
    root.appendChild(section);
}

export function makeButton(props) {
    const button = document.createElement("button");
    button.innerText = props.text;
    button.classList.add(props.cls);
    const parent = document.getElementById("buttonSection");
    parent.appendChild(button);
    if (props.cls !="operation"){
        button.addEventListener('click', () => {
            calculator.display(button.innerText)
            //calculator.updateDisplay()
        })
    } 
}

export function makeIcon(props) {
    const img = document.createElement("img");
    img.classList.add(props.cls);
    img.setAttribute("src", props.source);
    const parent = document.getElementById("topSection");
    parent.appendChild(img);
}
