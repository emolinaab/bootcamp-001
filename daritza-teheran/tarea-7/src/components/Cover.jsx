import "../styles/cover.css";
import { Masthead } from "./Masthead.jsx";
import { CovercardLine, CovercardImage, CovercardBottom } from "./Covercard.jsx";

const masthead = {
  title: "DOG WORLD",
  subtitle: "All about your furry friend and more",
};

const coverLines = [
  ["Caucasian Shepherd", "SPINONE", "Tibetan Mastiff"],
  ["All about", "COCKER SPANIELS"],
  ["+ Vet", "Lameness, a form of osteoarthritis"],
];

const coverImage = {
  url: "/src/assets/image.jpg",
  lines: ["+ Education", "What its behavior is so bad?"],
};

const coverBottom = {
  url: "/src/assets/codigo-de-barras.png",
  data: 'Dog world NÂ°301 - mensual $3.5',
};

export const Cover = () => {
  return (
    <main>
      <Masthead {...masthead} />
      <section className="cover-lines">
        {coverLines.map((clines) => (
          <CovercardLine lines={clines} />
        ))}
        <CovercardImage {...coverImage} />
      </section>
      <CovercardBottom {...coverBottom} />
    </main>
  );
};
