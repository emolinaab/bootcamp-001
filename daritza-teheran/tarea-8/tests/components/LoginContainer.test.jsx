import { render, screen } from "@testing-library/react";
import { Form } from "../../src/components/Form";
import { LoginContainer } from "../../src/components/LoginContainer";

describe("Testing in <LoginContainer />", () => {
  const logCont = <LoginContainer children={<Form />} />;

  test("Should make match with snapshot", () => {
    const { container } = render(logCont);
    expect(container).toMatchSnapshot();
  });

  test("Should show the form", () => {
    render(logCont);
    expect(screen.getByTestId("form")).toBeTruthy();
  });
});
