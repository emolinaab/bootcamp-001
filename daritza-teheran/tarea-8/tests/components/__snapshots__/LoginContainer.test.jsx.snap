// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <LoginContainer /> Should make match with snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1fk7jyc-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item 'box-shadow css-10emkex-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Sign in
      </h5>
      <form
        data-testid="form"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="email"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  placeholder="email@globant.com"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Email
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="password"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1p9d86b-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-1idn90j-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
